src/auth/src/credentials.rs

```diff
@@ -53,23 +54,21 @@ use std::future::Future;
 /// [Metadata Service]: https://cloud.google.com/compute/docs/metadata/overview
 /// [Google Compute Engine]: https://cloud.google.com/products/compute
 /// [Google Kubernetes Engine]: https://cloud.google.com/kubernetes-engine
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub struct Credential {
-    inner: Box<dyn traits::dynamic::Credential>,
+    inner: Arc<dyn traits::dynamic::Credential>,
```

Comment:
Future you may wonder why `Arc` and not `Rc` or `Box` consider a comment (I think `Rc` is out because you need `Send + Sync`, and `Box` because you need `Clone`).

Reply:
Done.